src/compose.ts

```diff
@@ -161,9 +162,29 @@ function normalizeService(service: Service, serviceNames: string[]): Service {
 		validateLabels(service.labels);
 	}
 
+	if (service.volumes) {
+		service.volumes.forEach((volume) => {
+			validateServiceVolume(volume, volumeNames);
+		});
+	}
+
 	return service;
 }
 
+function validateServiceVolume(serviceVolume: string, volumeNames: string[]) {
+	const colonIndex = serviceVolume.indexOf(':');
+	if (colonIndex === -1) {
+		throw new ValidationError(`Invalid volume: '${serviceVolume}'`);
+	}
+	const source = serviceVolume.slice(0, colonIndex);
+	if ([ '.', '/' ].indexOf(source[0]) !== -1) {
```

Comment:
I think `path.parse` might be good for this, because we can check the `.dir` parameter, and I think the code is a bit more clear.

Reply:
Good idea, updated.